submarineswap_server.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"log"
+
+	"swapper/mempoolspace"
+	"swapper/submarineswaprpc"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/lightningnetwork/lnd/zpay32"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Server is a sub-server of the main RPC server.
+type Server struct {
```

Comment:
I think it is better to move this file up to the root and leave only the proto and generated files in this folder